src/components/CoinsList/CoinsList.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import styles from './CoinsList.module.scss';
+import { Link, useLocation } from 'react-router-dom';
+import AppRoutes from '../../constants/routes';
+import { useAppDispatch, useAppSelector } from '../../hooks/reduxTypedHooks';
+import Checkbox from '../Checkbox';
+import {
+  addSelectedCoin,
+  removeSelectedCoin,
+} from '../../store/slices/coinsSelectedSlice';
+import { CoinTableProps } from './CoinsList.type';
+
+const { coin, coins, coin__name } = styles;
+
+const CoinsList: React.FC<CoinTableProps> = ({ coinsList }) => {
+  const location = useLocation();
+  const pageNumber = new URLSearchParams(location.search).get('page') || '1';
```

Comment:
Do you think something like the code below might have worked?
```
  const [searchParams] = useSearchParams();
  const pageNumber = searchParams.get('page') || '1';
```

Also I'd say that when using a framework, it's better to use its tools where possible other than the native URLSearchParams